Plugins/Flow.Launcher.Plugin.PluginsManager/PluginsManager.cs

```diff
@@ -0,0 +1,198 @@
+                return;
+
+            var filePath = Path.Combine(DataLocation.PluginsDirectory, $"{plugin.Name}{plugin.ID}.zip");
+            
+            try
+            {
+                Context.API.ShowMsg(Context.API.GetTranslation("plugin_pluginsmanager_downloading_plugin"),
+                                    Context.API.GetTranslation("plugin_pluginsmanager_please_wait"));
+
+                Utilities.Download(plugin.UrlDownload, filePath);
+
+                Context.API.ShowMsg(Context.API.GetTranslation("plugin_pluginsmanager_downloading_plugin"),
+                                    Context.API.GetTranslation("plugin_pluginsmanager_download_success"));
+            }
+            catch (Exception e)
+            {
+                Context.API.ShowMsg(Context.API.GetTranslation("plugin_pluginsmanager_downloading_plugin"),
+                                Context.API.GetTranslation("plugin_pluginsmanager_download_success"));
+
+                Log.Exception("PluginsManager", "An error occured while downloading plugin", e, "PluginDownload");
+            }
+
+            Application.Current.Dispatcher.Invoke(() => Install(plugin, filePath));
+        }
+
+        internal void Update()
+        {
+            throw new NotImplementedException();
+        }
+
+        internal bool PluginExists(string id)
+        {
+            return Context.API.GetAllPlugins().Any(x => x.Metadata.ID == id);
+        }
+
+        internal void PluginsManifestSiteOpen()
+        {
+            //Open from context menu https://git.vcmq.workers.dev/Flow-Launcher/Flow.Launcher.PluginsManifest
+            throw new NotImplementedException();
+        }
+
+        internal List<Result> Search(List<Result> results, string searchName)
+        {
+            if (string.IsNullOrEmpty(searchName))
+                return results;
+
+            return results
+                    .Where(x => StringMatcher.FuzzySearch(searchName, x.Title).IsSearchPrecisionScoreMet())
+                    .Select(x => x)
+                    .ToList();
+        }
+
+        internal List<Result> RequestInstallOrUpdate(string searchName)
+        {
+            var results = new List<Result>();
+
+            pluginsManifest
+                .UserPlugins
+                    .ForEach(x => results.Add(
+                        new Result
+                        {
+                            Title = $"{x.Name} by {x.Author}",
+                            SubTitle = x.Description,
+                            IcoPath = icoPath,
+                            Action = e =>
+                            {
+                                Application.Current.MainWindow.Hide();
+                                InstallOrUpdate(x);
+
+                                return true;
+                            }
+                        }));
+
+            return Search(results, searchName);
```

Comment:
Well I do think it will be more elegent to simply use the `Select` method to get the resultlist but this is fine, too.